### DISCUSSION TITLE
Filling Transparency with Checkerboard

### DISCUSSION THREAD

**Discussion by blue-j** [ImageMagick/ImageMagick#4551] [Help] [answered]
How to take an image, and its mask, and produce the output shown, filled with a checkerboard pattern?  Our old solution doesn't seem to work!

![source](https://user-images.githubusercontent.com/3990142/144768925-dfe446ba-2c34-4331-8a15-6d7dd82c85c3.jpg)
![mask](https://user-images.githubusercontent.com/3990142/144768928-6f1c1a17-0c06-490d-ba84-b17cff082d03.jpg)
![desired_output](https://user-images.githubusercontent.com/3990142/144768930-17312a8d-c817-43db-8110-aa07b1e81dde.jpg)

**Comment by fmw42** [comment]
What did you try? What version of ImageMagick? What OS/platform?

**Comment by fmw42** [comment]
Try this:

```
magick pattern:checkerboard -write mpr:tile +delete \
144768925-dfe446ba-2c34-4331-8a15-6d7dd82c85c3.jpg \
\( +clone -fill mpr:tile  -draw 'color 0,0 reset' \) \
144768928-6f1c1a17-0c06-490d-ba84-b17cff082d03.jpg \
-compose over -composite \
result1.jpg

```
![result1](https://user-images.githubusercontent.com/13458218/144769302-dcc493da-9c41-42d6-9dff-3f420b1a2cf4.jpg)

**Reply by fmw42** [reply]
This also works in ImageMagick 7

```
magick 144768925-dfe446ba-2c34-4331-8a15-6d7dd82c85c3.jpg \
-set option:dims "%wx%h" -size "%[dims]" tile:pattern:checkerboard \
144768928-6f1c1a17-0c06-490d-ba84-b17cff082d03.jpg -composite  result2.jpg

```
![result2](https://user-images.githubusercontent.com/13458218/144769481-71cd2915-12e3-445a-96ff-80cb41b08ca1.jpg)

**Comment by blue-j** [comment]
Thanks Fred!  Oh darn I made a mistake though.  The mask is inverted to what I provided... so how can I rewrite the command to work  with that?   I can't crack it!  I need the checkerboard to fill in the black area of the mask, not the white.  : )

**Reply by fmw42** [reply]
Negate the mask (in a parenthesis) before using it. But if you do that, then it will cover the main part of the image. So I do not understand what you want. Please clarify.

**Reply by blue-j** [reply]
Same source and output but using this mask:
![alpha_cha12](https://user-images.githubusercontent.com/3990142/144771498-94957fa9-cce9-4116-b24a-d33554d704b2.jpg)

**Reply by fmw42** [reply]
That is the same mask. If the checkerboard covers the image corresponding to the black, then it will cover all the real image and only leave the white area in the top right. Is that what you want?

**Reply by blue-j** [reply]
Just to cause even more confusion I uploaded the same mask the first time.  It has been replaced properly.

**Reply by fmw42** [reply]
Do you want the white or black to be checkerboard?

**Comment by fmw42** [comment, marked as answer]
Masks in composite always process the white area. So just invert your mask.

```
magick 144768925-dfe446ba-2c34-4331-8a15-6d7dd82c85c3.jpg \
-set option:dims "%wx%h" -size "%[dims]" tile:pattern:checkerboard \
\( 144771498-94957fa9-cce9-4116-b24a-d33554d704b2.jpg -negate \) -composite  result3.jpg



```

![result3](https://user-images.githubusercontent.com/13458218/144771932-019dd4e4-889a-4849-8381-60fa3b7292e1.jpg)

**Reply by blue-j** [reply]
Thanks so much!!  You are a wizard!!!!  : )
